### ISSUE TITLE
Content Pipeline cache gets invalidated

### ISSUE THREAD

**Issue by Apostolique** [MonoGame/MonoGame#7660]
I'm developing using vscode and omnisharp. I'm running into a case where during the project load, the MonoGame content builder is given a lower case drive letter. This makes this check fail and all content gets rebuilt:

https://github.com/MonoGame/MonoGame/blob/2850e2dfc98c4d7b85fb2494ecfec1424fea9add/MonoGame.Framework.Content.Pipeline/Builder/PipelineBuildEvent.cs#L190-L194

In the cached `.mgcontent` files, the SourceFile becomes something like:

```xml
<SourceFile>f:/SuperProject/Content/Test.png</SourceFile>
```

When I build my project though the drive letter is uppercase which makes the check above fail again and all content gets rebuilt again. When `.mgcontent` gets saved again, I see:

```xml
<SourceFile>F:/SuperProject/Content/Test.png</SourceFile>
```

Every time omnisharp restarts, I get lower case drive letter again. This makes my builds become really slow because the cache is always invalidated and all content gets rebuilt over and over.

Looks like the issue comes from: https://github.com/MonoGame/MonoGame/blob/2850e2dfc98c4d7b85fb2494ecfec1424fea9add/Tools/MonoGame.Content.Builder/BuildContent.cs#L182

The `Directory.GetCurrentDirectory()` isn't the true case sensitive path system.

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- DesktopGL

**Comment by Apostolique** [comment]
This seems related to https://github.com/microsoft/vscode/issues/43959.

**Comment by Apostolique** [comment]
Indirectly fixed by #7660.
